base/complex.jl

```diff
@@ -461,7 +461,7 @@ function ^{T<:AbstractFloat}(z::Complex{T}, p::Complex{T})
     if p==2 #square
         zr, zi = reim(z)
         x = (zr-zi)*(zr+zi)
-        y = 2zr*zi
+        y = T(2)*zr*zi
```

Comment:
Because of the type instability it creates for `Float16`:
```julia
julia> typeof(2*Float16(1.0))
Float32
```

Maybe better write it as `T(2*zr*zi)`?

(I wonder whether that promotion exception for `Float16` is really a good idea...)